server/authorization.go

```diff
@@ -495,38 +602,13 @@ func (a *AuthorizationServer) corsPreflightResponse(
 	api string) *authv3.CheckResponse {
 
 	log.Debugf("sending cors preflight for api: %v", envRequest.GetAPISpec().ID)
-	return a.createEnvoyDenied(envRequest.Request, envRequest, tracker, authContext, api, rpc.CANCELLED, typev3.StatusCode_NoContent)
-}
-
-func (a *AuthorizationServer) notFound(req *authv3.CheckRequest, envRequest *config.EnvironmentSpecRequest,
-	tracker *prometheusRequestMetricTracker, api string) *authv3.CheckResponse {
-	return a.createConditionalEnvoyDenied(req, envRequest, tracker, nil, api, rpc.NOT_FOUND)
-}
-
-func (a *AuthorizationServer) unauthenticated(req *authv3.CheckRequest, envRequest *config.EnvironmentSpecRequest,
-	tracker *prometheusRequestMetricTracker, api string) *authv3.CheckResponse {
-	return a.createConditionalEnvoyDenied(req, envRequest, tracker, nil, api, rpc.UNAUTHENTICATED)
-}
-
-func (a *AuthorizationServer) unavailable(req *authv3.CheckRequest) *authv3.CheckResponse {
-	log.Errorf("sending service unavailable")
-	return a.createConditionalEnvoyDenied(req, nil, nil, nil, "", rpc.UNAVAILABLE)
-}
-
-func (a *AuthorizationServer) internalError(req *authv3.CheckRequest, envRequest *config.EnvironmentSpecRequest,
-	tracker *prometheusRequestMetricTracker, err error) *authv3.CheckResponse {
-	log.Errorf("sending internal error: %v", err)
-	return a.createConditionalEnvoyDenied(req, envRequest, tracker, nil, "", rpc.INTERNAL)
-}
-
-func (a *AuthorizationServer) denied(req *authv3.CheckRequest, envRequest *config.EnvironmentSpecRequest,
-	tracker *prometheusRequestMetricTracker, authContext *auth.Context, api string) *authv3.CheckResponse {
-	return a.createConditionalEnvoyDenied(req, envRequest, tracker, authContext, api, rpc.PERMISSION_DENIED)
+	return a.createEnvoyDenied(envRequest.Request, envRequest, tracker, authContext, api, fault.CreateAdapterFault("", rpc.CANCELLED, typev3.StatusCode_NoContent))
 }
 
-func (a *AuthorizationServer) quotaExceeded(req *authv3.CheckRequest, envRequest *config.EnvironmentSpecRequest,
-	tracker *prometheusRequestMetricTracker, authContext *auth.Context, api string) *authv3.CheckResponse {
-	return a.createConditionalEnvoyDenied(req, envRequest, tracker, authContext, api, rpc.RESOURCE_EXHAUSTED)
+func (a *AuthorizationServer) handleFault(req *authv3.CheckRequest, envRequest *config.EnvironmentSpecRequest,
+	tracker *prometheusRequestMetricTracker, api string, authContext *auth.Context, adapterFault *fault.AdapterFault) *authv3.CheckResponse {
```

Comment:
The `adapterFault` might be better off passed in here as an `error`, and you can test casting it to `fault.AdapterFault` here, or in the `createConditionalEnvoyDenied`. WDYT?

Reply:
I had used error initially but changed it later on due to misunderstanding an earlier comment. Changing it back to error. Thanks!